core/src/test/scala/unit/kafka/server/metadata/BrokerMetadataListenerTest.scala

```diff
@@ -1,413 +0,0 @@
-        endOffset,
-        0,
-        util.Arrays.asList(
-          new ApiMessageAndVersion(new PartitionChangeRecord().
-            setPartitionId(0).
-            setTopicId(BAR_ID).
-            setRemovingReplicas(Collections.singletonList(1)), 0.toShort),
-          new ApiMessageAndVersion(new PartitionChangeRecord().
-            setPartitionId(0).
-            setTopicId(BAR_ID).
-            setRemovingReplicas(Collections.emptyList()), 0.toShort)
-        )
-      )
-    )
-    listener.getImageRecords().get()
-  }
-
-  @Test
-  def testHandleCommitsWithNoSnapshotterDefined(): Unit = {
-    val listener = newBrokerMetadataListener(maxBytesBetweenSnapshots = 1000L)
-    try {
-      val brokerIds = 0 to 3
-
-      registerBrokers(listener, brokerIds, endOffset = 100L)
-      createTopicWithOnePartition(listener, replicas = brokerIds, endOffset = 200L)
-      listener.getImageRecords().get()
-      assertEquals(200L, listener.highestMetadataOffset)
-
-      generateManyRecords(listener, endOffset = 1000L)
-      assertEquals(1000L, listener.highestMetadataOffset)
-    } finally {
-      listener.close()
-    }
-  }
-
-  @Test
-  def testCreateSnapshot(): Unit = {
-    val snapshotter = new MockMetadataSnapshotter()
-    val listener = newBrokerMetadataListener(snapshotter = Some(snapshotter),
-      maxBytesBetweenSnapshots = 1000L)
-    try {
-      val brokerIds = 0 to 3
-
-      registerBrokers(listener, brokerIds, endOffset = 100L)
-      createTopicWithOnePartition(listener, replicas = brokerIds, endOffset = 200L)
-      listener.getImageRecords().get()
-      assertEquals(200L, listener.highestMetadataOffset)
-
-      // Check that we generate at least one snapshot once we see enough records.
-      assertEquals(-1L, snapshotter.prevCommittedOffset)
-      generateManyRecords(listener, 1000L)
-      assertEquals(1000L, snapshotter.prevCommittedOffset)
-      assertEquals(1000L, snapshotter.activeSnapshotOffset)
-      snapshotter.activeSnapshotOffset = -1L
-
-      // Test creating a new snapshot after publishing it.
-      val publisher = new MockMetadataPublisher()
-      listener.startPublishing(publisher).get()
-      generateManyRecords(listener, 2000L)
-      listener.getImageRecords().get()
-      assertEquals(2000L, snapshotter.activeSnapshotOffset)
-      assertEquals(2000L, snapshotter.prevCommittedOffset)
-
-      // Test how we handle the snapshotter returning false.
-      generateManyRecords(listener, 3000L)
-      assertEquals(2000L, snapshotter.activeSnapshotOffset)
-      generateManyRecords(listener, 4000L)
-      assertEquals(2000L, snapshotter.activeSnapshotOffset)
-      snapshotter.activeSnapshotOffset = -1L
-      generateManyRecords(listener, 5000L)
-      assertEquals(5000L, snapshotter.activeSnapshotOffset)
-      assertEquals(null, snapshotter.failure.get())
-    } finally {
-      listener.close()
-    }
-  }
-
-  @Test
-  def testNotSnapshotAfterMetadataVersionChangeBeforePublishing(): Unit = {
```

Comment:
Are all of these tests recreated somewhere?

Reply:
in the latest PR, `MetadataLoaderTest.java` replaces `BrokerMetadataListenerTest.scala`